components/definitions/today.d.ts

```diff
@@ -0,0 +1,107 @@
+declare namespace BraveToday {
+
+  // Messages
+  namespace Messages {
+    // getFeed
+    export type GetFeedResponse = {
+      feed: BraveToday.Feed | undefined
+    }
+    // getPublishers
+    export type GetPublishersResponse = {
+      publishers: BraveToday.Publishers | undefined
+    }
+    // getImageData
+    export type GetImageDataPayload = {
+      url: string
+    }
+    export type GetImageDataResponse = {
+      dataUrl: string
+    }
+    // set publisher prefs
+    export type SetPublisherPrefPayload = {
+      publisherId: string
+      // Boolean for explicit change, null for remove pref.
+      enabled: boolean | null
+    }
+    export type SetPublisherPrefResponse = GetPublishersResponse
+    export type ClearPrefsPayload = {}
+    export type ClearPrefsResponse = SetPublisherPrefResponse
+    // isFeedUpdateAvailable
+    export type IsFeedUpdateAvailablePayload = {
+      hash: string
+    }
+    export type IsFeedUpdateAvailableResponse = {
+      isUpdateAvailable: boolean
+    }
+  }
+
+  export type FeedItem = (Article | Deal)
+
+  export interface Feed {
+    hash: string
+    pages: Page[]
+    featuredSponsor?: Article
+    featuredArticle?: Article
+    featuredDeals?: Deal[]
+  }
+
+  export interface Page {
+    articles: Article[] // 13
```

Comment:
What are the numbers here? The maximum number of entries?

Reply:
It was just a helpful indicator to show how many are expected to be put in each of those arrays, since you can't easily specify array size in typescript (even if we wanted to). I can make it clearer.